src/dsp_tools/commands/ingest_xmlupload/user_information.py

```diff
@@ -0,0 +1,120 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import pandas as pd
+
+separator = "\n    "
+list_separator = "\n    - "
+
+
+@dataclass(frozen=True)
+class IngestInformation:
+    """
+    This class stores the information about the mapping of ids provided by the dsp-ingest service
+    and the filepaths used in the XML file.
+    """
+
+    unused_mediafiles: list[str]
+    mediafiles_no_id: list[tuple[str, str]]
+    maximum_prints: int = 20
+    csv_directory_path: Path = field(default=Path.cwd())
+    unused_mediafiles_csv: str = "UnusedMediaUploadedInSipi.csv"
+    mediafiles_no_id_csv: str = "FilesNotUploadedToSipi.csv"
+
+    def ok_msg(self) -> str | None:
+        """
+        This function checks if no media was unused or not uploaded.
+        If that is the case it returns the message,
+        if not, it ends without an effect.
+
+        Returns:
+            Message if all went well.
+        """
+        if not self.unused_mediafiles and not self.mediafiles_no_id:
+            return (
+                "All multimedia files referenced in the XML file were uploaded through dsp-ingest.\n"
+                "No multimedia files were uploaded through dsp-ingest that were not referenced in the XML file."
+            )
+        return None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function generates the user message and saves a file with the information
+        if a lot of resources are affected.
+
+        Returns:
+            User message
+        """
+        self._save_csv_if_applicable()
+        return self._get_error_msg()
+
+    def _get_error_msg(self) -> str:
+        msg_list = [
+            "The upload cannot continue as there are problems with the multimedia files referenced in the XML.",
+        ]
+        if has_msg := self._get_unused_mediafiles_msg():
+            msg_list.append(has_msg)
+        if has_msg := self._get_mediafiles_no_id_msg():
+            msg_list.append(has_msg)
+        return separator.join(msg_list)
+
+    def _get_mediafiles_no_id_msg(self) -> str | None:
+        if 0 < len(self.mediafiles_no_id) <= self.maximum_prints:
+            return (
+                "The data XML file contains references to the following multimedia files "
+                "which were not previously uploaded through dsp-ingest:"
+                + list_separator
+                + list_separator.join([f"Resource ID: '{x[0]}' | Filepath: '{x[1]}'" for x in self.mediafiles_no_id])
+            )
+        elif len(self.mediafiles_no_id) > self.maximum_prints:
+            return (
+                "The data XML file contains references to multimedia files "
+                "which were not previously uploaded through dsp-ingest:\n"
+                f"    The file with the resource IDs and problematic filenames was saved at "
+                f"'{self.csv_directory_path}/{self.mediafiles_no_id_csv}'."
```

Comment:
This way of concatenating a Path object with a "/" to a string is not recommended. "/" might be the correct separator on UNIX, but not on windows. You can let pathlib do the work for you, it will always choose the right separator: 

```suggestion
                f"'{self.csv_directory_path / self.mediafiles_no_id_csv}'."
```

Reply:
But thinking about your variable self.csv_directory_path: I think it's redundant. If you save a file, it will always be saved in the cwd. I believe you can completely remove the variable self.csv_directory_path from this class.

See this code example:
```python
>>> assert Path("foo").absolute() == Path.cwd() / "foo"
```

Reply:
It served two purposes for me. If the path changed it is easy to modify at only one place.
It is easier for testing. This way I don't have to make these over complicated path patches, but can just overwrite it.

Reply:
okay, I didn't know that, that's a good reason